### PR TITLE
Throttler code: some refactoring and cleanup

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

This is a quick followup to https://github.com/vitessio/vitess/pull/15988. This PR includes no logic changes. It's all about changing some names, and moving some files. Specifically, removing "MySQL" from a lot of function and variable names, where it's just irrelevant. Removing the `mysql` package as it has nothing specific to MySQL and converging it into the `throttle/base` package.

Other than the refactor of names, the tests are unchanges and again no logic changes were made.

## Related Issue(s)

https://github.com/vitessio/vitess/issues/15624

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/throttle/base/tablet_results_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (rohit-nayak-ps) [inline go/vt/vttablet/tabletserver/throttle/base/tablet_results_test.go:2]: Should be 2024?

File: go/vt/vttablet/tabletserver/throttle/base/tablet_results.go [hunk 2]
```diff
@@ -14,63 +14,42 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// This codebase originates from https://github.com/github/freno, See https://github.com/github/freno/blob/master/LICENSE
```
Comment (rohit-nayak-ps) [inline go/vt/vttablet/tabletserver/throttle/base/tablet_results.go:17]: No need for this attribution anymore?

**Review by rohit-nayak-ps — APPROVED** [review]

File: go/vt/vttablet/tabletserver/throttle/base/tablet_results_test.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/vttablet/tabletserver/throttle/base/tablet_results_test.go:2]: fixed

File: go/vt/vttablet/tabletserver/throttle/base/tablet_results.go (hunk 2 above)
Comment (shlomi-noach) [reply go/vt/vttablet/tabletserver/throttle/base/tablet_results.go:17]: Not in this specific file. The code in this file is not derived from `freno`, and the note was needlessly copy+pasted from another file.

**Review by mattlord — APPROVED** [review]

**Pull request merged**